components/google_sheets/actions/add-conditional-format-rule/add-conditional-format-rule.mjs

```diff
@@ -0,0 +1,492 @@
+      type: "string",
+      label: "Vertical Align",
+      description: "The vertical alignment",
+      options: [
+        "TOP",
+        "MIDDLE",
+        "BOTTOM",
+      ],
+      optional: true,
+    },
+    wrapStrategy: {
+      type: "string",
+      label: "Wrap Strategy",
+      description: "The wrap strategy",
+      options: [
+        "OVERFLOW_CELL",
+        "LEGACY_WRAP",
+        "CLIP",
+        "WRAP",
+      ],
+      optional: true,
+    },
+    textDirection: {
+      type: "string",
+      label: "Text Direction",
+      description: "The text direction",
+      options: [
+        "LEFT_TO_RIGHT",
+        "RIGHT_TO_LEFT",
+      ],
+      optional: true,
+    },
+    textFormat: {
+      type: "object",
+      label: "Text Format",
+      description: "The text format options",
+      optional: true,
+    },
+    fontFamily: {
+      type: "string",
+      label: "Font Family",
+      description: "The font family",
+      optional: true,
+    },
+    fontSize: {
+      type: "integer",
+      label: "Font Size",
+      description: "The font size",
+      optional: true,
+    },
+    bold: {
+      type: "boolean",
+      label: "Bold",
+      description: "Whether the text is bold",
+      optional: true,
+    },
+    italic: {
+      type: "boolean",
+      label: "Italic",
+      description: "Whether the text is italic",
+      optional: true,
+    },
+    underline: {
+      type: "boolean",
+      label: "Underline",
+      description: "Whether the text is underlined",
+      optional: true,
+    },
+    strikethrough: {
+      type: "boolean",
+      label: "Strikethrough",
+      description: "Whether the text is strikethrough",
+      optional: true,
+    },
+    url: {
+      type: "string",
+      label: "URL",
+      description: "The URL for the link",
+      optional: true,
+    },
+    hyperlinkDisplayType: {
+      type: "string",
+      label: "Hyperlink Display Type",
+      description: "The hyperlink display type",
+      options: [
+        "PLAIN_TEXT",
+        "LINKED",
+      ],
+      optional: true,
+    },
+    InterpolationPointType: {
+      type: "string",
+      label: "Interpolation Point Type",
+      description: "The interpolation point type",
+      options: [
+        "MIN",
+        "MAX",
+        "NUMBER",
+        "PERCENT",
+        "PERCENTILE",
+      ],
+      optional: true,
+    },
+    index: {
+      type: "integer",
+      label: "Index",
+      description: "The zero-based index of the rule",
+    },
+  },
+  async run() {
+    const {
+      startCol,
+      endCol,
+      startRow,
+      endRow,
+    } = this.props.googleSheets._parseRangeString(`${this.props.worksheetId}!${this.props.range}`);
+
+    const rule = {
+      range: {
+        sheetId: this.props.worksheetId,
+        startRowIndex: startRow,
+        endRowIndex: endRow,
+        startColumnIndex: startCol.charCodeAt(0) - 65,
+        endColumnIndex: endCol.charCodeAt(0) - 64,
+      },
+    };
```

Comment:
Props can be referenced as `this.<propName>`, so worksheetId would be `this.worksheetId`, range would be `this.range`, etc. Applies throughout this PR.
```suggestion
    const {
      startCol,
      endCol,
      startRow,
      endRow,
    } = this.googleSheets._parseRangeString(`${this.worksheetId}!${this.range}`);

    const rule = {
      range: {
        sheetId: this.worksheetId,
        startRowIndex: startRow,
        endRowIndex: endRow,
        startColumnIndex: startCol.charCodeAt(0) - 65,
        endColumnIndex: endCol.charCodeAt(0) - 64,
      },
    };
```